### DISCUSSION TITLE
Get a list of ports of an S-parameter

### DISCUSSION THREAD

**Discussion by giuseppe1692** [ansys/pyaedt#4724] [Q&A]
Hi all, 
I have a question regarding how to obtain a list of all the port of an S-parameter imported using PyAedt scripts:
I wrote this code and it works (by using this code I import an S parameter and add the PageConnectors):

```python
def carica_snp(path, offset_x, offset_y):
    aedt_app = pyaedt.Circuit(projectname=nome_progetto, designname=nome_design)
    aedt_app.modeler.schematic.schematic_units = "mil"
    mlo_component = aedt_app.modeler.components.create_touchstone_component(model_name= path, location=[offset_x, offset_y])
    aedt_app.oeditor.AddPinPageconnectors(["NAME:Selections","Selections:=",aedt_app.oeditor.GetSelections()])
```  
Now, I want to obtain a list of the name of all the ports on the S-parameter and add GND if the name of the port has a certain name. 

Thank you in advance.

**Comment by mahesh-dhanekula** [comment]
@giuseppe1692 
Below snippet will let you read pin information which are nothing but ports of your touchstone model
```
for pin in mlo_component.pins:
    pin.name # Pin Name
    pin.location # Pin Location (x, y)
```
